tests/engine/effect_test.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+from tests.mocks import world_mock
+from scripts.engine import world
+from scripts.engine.core.constants import AfflictionTrigger
+from scripts.nqp.actions.afflictions import Affliction
+from scripts.engine.component import Afflictions, Identity, Position, Knowledge
+from scripts.engine.effect import TriggerAfflictionsEffect, ReduceSkillCooldownEffect
+from scripts.nqp.actions.skills import Move
+
+
+class MockAffliction(Affliction):
+    name = 'mock_affliction'
+    identity_tags = [
+        "affect_stat"
+    ]
+    triggers = [
+        "movement"
+    ]
+    required_tags = [
+        "other_entity"
+    ]
+
+    def __init__(self, creator, affected_entity, duration):
+        super().__init__(creator, affected_entity, duration)
+        self.triggered = False
+
+    def build_effects(self, entity):
+        self.triggered = True
+        return []
+
+
+class MockAfflictionDamage(MockAffliction):
+    triggers = [
+        "take_damage"
+    ]
+
+
+class MockAfflictionMovement(MockAffliction):
+    triggers = [
+        "movement"
+    ]
+
+
+class TestEffects:
+
+    @staticmethod
+    def _create_default_entity():
+        components = [Identity('mock_entity'), Position(0, 0)]
+        entity = world.create_entity(components)
+        return entity
+
+    def test_trigger_afflictions_effect(self):
+        """
+        Test for the trigger afflictions effect
+        """
+        affliction_called = None
+
+        def _trigger_affliction_mock(affliction):
+            nonlocal affliction_called
+            affliction_called = affliction
+
+        world_mock.mock_methods({'apply_affliction': _trigger_affliction_mock})
```

Comment:
Here if we didn't mock world.apply_affliction it would throw an Exception because the gamemap isn't loaded. However we circumvent this by mocking it with the method above since we are only interested in knowing if it gets called.